src/Columns/ColumnFunction.cpp

```diff
@@ -59,6 +61,34 @@ ColumnPtr ColumnFunction::cut(size_t start, size_t length) const
     return ColumnFunction::create(length, function, capture, is_short_circuit_argument, is_function_compiled);
 }
 
+void ColumnFunction::insertFrom(const IColumn & src, size_t n)
+{
+    const ColumnFunction & src_func = assert_cast<const ColumnFunction &>(src);
+
+    for (size_t i = 0, size = captured_columns.size(); i < size; ++i)
+    {
+        auto mut_column = IColumn::mutate(std::move(captured_columns[i].column));
+        mut_column->insertFrom(*src_func.captured_columns[i].column, n);
```

Comment:
Should not we check that src_func.captured_columns size equals to our captured_columns size ?

Reply:
Also this function can just use `insertRangeFrom` with start = n and length = 1.

Reply:
> Should not we check that src_func.captured_columns size equals to our captured_columns size ?

Yes, I will add assert.

> Also this function can just use insertRangeFrom with start = n and length = 1.

`insertRangeFrom` can be more heavy for some columns, so I've implemented a separate method.